src/sqlfluff/core/templaters/jinja.py

```diff
@@ -286,6 +286,10 @@ def get_context(self, fname=None, config=None, **kw) -> Dict:
         live_context = super().get_context(fname=fname, config=config)
         # Apply dbt builtin functions if we're allowed.
         if config:
+            # first make libraries available in the context
+            # so they can be used by the macros too
+            live_context.update(self._extract_libraries_from_config(config=config))
```

Comment:
I was wondering if changing the order could cause issues for other users, e.g. if a library module depends on a macro. I don't _think_ that could happen, though, because the modules are just Python; they don't work with a Jinja2 `Environment`.

Reply:
As I understand it the library definition do represent macro's, but as part of an isolated package. So I don't think there is an option where a library refers to a macro that exists outside of the scope of the library.

Reply:
(thinking about `dbt_utils` here which is a library, but contains macros)